invesalius/gui/task_navigator.py

```diff
@@ -1132,14 +1151,46 @@ def OnLinkLoad(self, event=None):
                 self.obj_name = data[0][1]
                 self.obj_ref_mode = int(data[0][-1])
 
+                if os.path.exists(self.obj_name):
```

Comment:
This section of loading the file should preferably not be here. So, far the STL file has been loaded elsewhere, and if possible we should keep it outside task_navigator.